src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -316,11 +325,18 @@ public function updateAction(Request $request)
                 return $this->redirectHandler->redirectToResource($configuration, $resource);
             }
 
-            if ($configuration->hasStateMachine()) {
-                $this->stateMachine->apply($configuration, $resource);
+            try {
+                $this->resourceUpdater->applyTransitionAndFlush($resource, $configuration, $this->manager);
+            } catch (RaceConditionException $exception) {
+                if (!$configuration->isHtmlRequest()) {
+                    return $this->viewHandler->handle($configuration, View::create($form, Response::HTTP_BAD_REQUEST));
```

Comment:
think i prefer 409!
https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.10

> Conflicts are most likely to occur in response to a PUT request. For example, if versioning were being used and the entity being PUT included changes to a resource which conflict with those made by an earlier (third-party) request, the server might use the 409 response to indicate that it can't complete the request.